src-frontend/app/controllers/procedure.js

```diff
@@ -63,6 +64,29 @@ export default Ember.ObjectController.extend({
             });
 
             this.endPropertyChanges();
+        },
+        generateProtocol: function() {
+            var filename = this.get('model.title') + '.xml';
+            $.ajax({
+                beforeSend: function(xhr) {
+                    xhr.setRequestHeader('Authorization', 'Token ' + Cookies.get('authorizationToken'));
+                },
+                type: 'GET',
+                url: ENV.APP.API_HOST + '/' + ENV.APP.API_NAMESPACE + '/procedures/' + this.get('model.id') + '/generate/',
+                success: function(data) {
+                    var blob = new Blob([(new XMLSerializer()).serializeToString(data)], { type: 'text/xml' });
```

Comment:
Still very hacky but make sure you create an issue to fix it later and make a comment about it here.